lib/tests/modules/adhoc-freeformType-survives-type-merge.nix

```diff
@@ -6,10 +6,10 @@
   };
   freeformType =
     let
-      a = lib.types.attrsOf (lib.types.submodule { options.bar = lib.mkOption { }; });
```

Comment:
This only worked previously because the freeformType wasn't handled by means of options.
Now that this is consistent. The test broke, because `types.submodule` always re-creates the type; even if there is only one.